src/lib/UI/Service/ContentTypeIconResolver.php

```diff
@@ -0,0 +1,84 @@
+<?php
+
+/**
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+declare(strict_types=1);
+
+namespace EzSystems\EzPlatformAdminUi\UI\Service;
+
+use eZ\Publish\API\Repository\Values\ContentType\ContentType;
+use eZ\Publish\Core\MVC\ConfigResolverInterface;
+use Symfony\Component\Asset\Packages;
+
+class ContentTypeIconResolver
+{
+    /** @var \eZ\Publish\Core\MVC\ConfigResolverInterface */
+    private $configResolver;
+
+    /** @var \Symfony\Component\Asset\Packages */
+    private $packages;
+
+    /** @var string|null */
+    private $defaultThumbnail;
+
+    /**
+     * @param \eZ\Publish\Core\MVC\ConfigResolverInterface $configResolver
+     * @param \Symfony\Component\Asset\Packages $packages
+     * @param string|null $defaultThumbnail
+     */
+    public function __construct(ConfigResolverInterface $configResolver, Packages $packages, string $defaultThumbnail)
+    {
+        $this->configResolver = $configResolver;
+        $this->packages = $packages;
+        $this->defaultThumbnail = $defaultThumbnail;
+    }
+
+    /**
+     * Returns path to content type icon.
+     *
+     * Path is resolved based on configuration (ezpublish.system.<SCOPE>.content_type). If there isn't coresponding
+     * entry for given content type, then path to default icon will be returned.
+     *
+     * @param \eZ\Publish\API\Repository\Values\ContentType\ContentType|string $contentType
+     *
+     * @return string|null
+     */
+    public function getContentTypeIcon($contentType): ?string
+    {
+        if ($contentType instanceof ContentType) {
+            $contentType = $contentType->identifier;
+        }
+
+        $thumbnail = null;
```

Comment:
Here could be `$thumbnail = $this->defaultThumbnail;` and the `if empty($thumbnail)` is not needed.

Reply:
Not really. If you configure null or empty string as the content type icon then default icon should be used.